src/BackedEnum.php

```diff
@@ -28,17 +28,17 @@ public static function fromName(string $fqn, string $keyName): \BackedEnum
      */
     public static function tryFromName(string $fqn, string $keyName): ?\BackedEnum
     {
-        if (is_a($fqn, \BackedEnum::class, true) === false) {
-            throw new InvalidArgumentException('It is only possible to get names of backedEnums, "' . $fqn . '" provided');
+        if (!is_a($fqn, \BackedEnum::class, true)) {
```

Comment:
Although in this case, the function always returns a boolean, native PHP functions are notorious for returning mixed types. the NOT operator (!) does not only check against a type, but also casts the value implicitly to a boolean, so if the return value of a function is `0` or `[]` it will also result in the exception being thrown. As it is complex to explain in code reviews when a NOT operator is allowed and when it is not 😛, I try to give a good example and never use it at all, even though in this case it would be possible to use it without causing any unwanted bugs.

Reply:
Done 🍏

Reply:
There is a solution for this problem: https://github.com/thecodingmachine/safe